File: ts-sdk/src/sdk/models/shared/walletwithbalances.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { AssetHolder } from "./assetholder";


export class WalletWithBalancesBalances extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=main" })
  main: AssetHolder;
}

export class WalletWithBalances extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=balances" })
  balances: WalletWithBalancesBalances;

  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=ledger" })
  ledger: string;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;
}

File: python-sdk/src/formanceapi/models/shared/walletwithbalances.py
import dataclasses
import dateutil.parser
from ..shared import assetholder as shared_assetholder
from dataclasses_json import dataclass_json
from datetime import datetime
from formanceapi import utils
from marshmallow import fields
from typing import Any


@dataclass_json
@dataclasses.dataclass
class WalletWithBalancesBalances:
    main: shared_assetholder.AssetHolder = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('main') }})
    

@dataclass_json
@dataclasses.dataclass
class WalletWithBalances:
    balances: WalletWithBalancesBalances = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('balances') }})
    created_at: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdAt'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    ledger: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ledger') }})
    metadata: dict[str, Any] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    